huaweicloud/services/cph/resource_huaweicloud_cph_server.go

```diff
@@ -0,0 +1,802 @@
+			},
+			"auto_renew": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: `Whether auto renew is enabled. Valid values are "true" and "false".`,
+			},
+			"availability_zone": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Computed:    true,
+				Description: `The name of the AZ where the cloud server is located.`,
+			},
+			"keypair_name": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The key pair name, which is used for logging in to the cloud phone through ADB.`,
+			},
+			"enterprise_project_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `the enterprise project ID.`,
+			},
+			"ports": {
+				Type:        schema.TypeList,
+				Elem:        cphServerApplicationPortSchema(),
+				Optional:    true,
+				Computed:    true,
+				ForceNew:    true,
+				Description: `The application port enabled by the cloud phone.`,
+			},
+			"order_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: `The order ID.`,
+			},
+			"addresses": {
+				Type:        schema.TypeList,
+				Elem:        cphServerAddressSchema(),
+				Computed:    true,
+				Description: `The IP addresses of the CPH server.`,
+			},
+			"status": {
+				Type:        schema.TypeInt,
+				Computed:    true,
+				Description: `The CPH server status.`,
+			},
+		},
+	}
+}
+
+func cphServerBandWidthSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"bandwidth_share_type": {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: `The bandwidth type.`,
+				ValidateFunc: validation.StringInSlice([]string{
+					"0", "1",
+				}, false),
+			},
+			"bandwidth_id": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `The bandwidth ID.`,
+			},
+			"bandwidth_size": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Computed:     true,
+				Description:  `The bandwidth (Mbit/s).`,
+				ValidateFunc: validation.IntBetween(1, 2000),
+			},
+			"bandwidth_charge_mode": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+				Description: `Which the bandwidth used by the CPH server is billed.`,
+				ValidateFunc: validation.StringInSlice([]string{
+					"0", "1",
+				}, false),
+			},
+		},
+	}
+	return &sc
+}
+
+func cphServerPublicIpSchema() *schema.Resource {
+	sc := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"eip": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Elem:        cphServerPublicIpEipSchema(),
+				Optional:    true,
+				Computed:    true,
+				Description: `The configuration of the new EIP. The system automatically assign an EIP.`,
+			},
+			"ids": {
```

Comment:
how many nics and eips can the user buy?

Reply:
only one, more detail please see document. 
it's consistent with the api.

Reply:
change to eip_id